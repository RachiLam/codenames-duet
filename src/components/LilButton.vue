<template>
  <button
    :class="classes"
    @click="$emit('click')"
  />
</template>

<script>
export default {
  props: {
    activated: {
      type: Boolean,
      required: true,
    },
    type: {
      type: String,
      default: 'grey',
      validator: (v) => ['green', 'grey'].indexOf(v) >= 0,
    },
  },
  computed: {
    classes() {
      return {
        [this.type]: true,
        activated: this.activated,
      };
    },
  },
};
</script>

<style lang="scss" scoped>
  @use 'styles/colors';
  @use 'styles/constants';

  button {
    width: constants.$lil-button-size;
    height: constants.$lil-button-size;
    background-color: colors.$accent-light;
    border-radius: 25%;

    &.activated {
      &.green {
        background-color: colors.$success;
      }

      &.grey {
        background-color: colors.$accent-dark;
      }
    }
  }
</style>
