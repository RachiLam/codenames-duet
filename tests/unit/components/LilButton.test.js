import LilButton from 'src/components/LilButton.vue';

const { shallowMount } = getMountUtils({
  component: LilButton,
});

describe('components/LilButton', () => {
  describe('when not activated', () => {
    it('renders as inactive', () => {
      const wrapper = shallowMount({
        props: {
          activated: false,
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('when activated', () => {
    it('renders as activated', () => {
      const wrapper = shallowMount({
        props: {
          activated: true,
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('when green', () => {
    it('renders as green', () => {
      const wrapper = shallowMount({
        props: {
          type: 'green',
          activated: false,
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('when grey', () => {
    it('renders as grey', () => {
      const wrapper = shallowMount({
        props: {
          type: 'grey',
          activated: false,
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
