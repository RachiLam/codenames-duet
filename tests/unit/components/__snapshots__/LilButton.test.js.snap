// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/LilButton when activated renders as activated 1`] = `
<button
  class="grey activated"
/>
`;

exports[`components/LilButton when green renders as green 1`] = `
<button
  class="green"
/>
`;

exports[`components/LilButton when grey renders as grey 1`] = `
<button
  class="grey"
/>
`;

exports[`components/LilButton when not activated renders as inactive 1`] = `
<button
  class="grey"
/>
`;
